include/coroutine.h

```diff
@@ -0,0 +1,52 @@
+#ifndef _COROUTINE_H
+#define	_COROUTINE_H
+
+#include "coroutine_impl.h"
+#include "sys/stdtypes.h"
+#include <bits/types/struct_timespec.h>
+#include <sys/ucontext.h>
+typedef struct uzfs_coroutine uzfs_coroutine_t;
+
+extern void coroutine_init(void);
+extern void coroutine_fini(void);
+extern void coroutine_key_create(uint32_t *idx);
+extern void* coroutine_getkey(uint32_t idx);
+extern int coroutine_setkey(uint32_t idx, void *local);
```

Comment:
uniform the naming style for "key_create", and it's better to add a comment to document why there is no operation for deletion.